### PR TITLE
[move-examples] Added more Move examples

### PR DESCRIPTION
**Pull request by junkil-park**
Added the following examples to
`language/documentation/examples/experiment`:
- basic-coin: extracted from the Move tutorial
- coin-swap: Uniswap like contract in Move
- rounding-error: examples of verifying rounding errors using Prover
- math-puzzle: using Prover for fun

TODO:
- The next step is to add more document to the examples (e.g,
  README.md, code comments, ...).

## Motivation

To add more Move examples

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

cargo test

### CODE REVIEW

**Review by wrwg — APPROVED** [review]
Nice!

File: language/documentation/examples/experimental/rounding-error/sources/Reserve.move [hunk 1]
```diff
@@ -0,0 +1,89 @@
+module NamedAddr::Reserve {
```
Comment (DavidLDill) [inline language/documentation/examples/experimental/rounding-error/sources/Reserve.move:1]: Comment explaining what this demonstrates.  If it comes from a real contract, which one?  (I don't think it does in this case).

File: language/documentation/examples/experimental/rounding-error/sources/Reserve.move [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        let coin1info = borrow_global_mut<Coin1Info>(ADMIN);
+        let coin2_amount_to_reserve = FixedPoint32::multiply_u64(amount_to_mint, *& coin1info.reserve_coin2.backing_ratio) + 1;
+        assert!(backing_coin2 == coin2_amount_to_reserve, 2);
+        coin1info.reserve_coin2.backing_value = coin1info.reserve_coin2.backing_value + coin2_amount_to_reserve;
+        coin1info.total_value = coin1info.total_value + amount_to_mint;
+        amount_to_mint
+    }
+    spec mint_coin1 {
+        let coin1info = global<Coin1Info>(ADMIN);
+        let coin2_amount_to_reserve = FixedPoint32::spec_multiply_u64(amount_to_mint, coin1info.reserve_coin2.backing_ratio) + 1;
```
Comment (DavidLDill) [inline language/documentation/examples/experimental/rounding-error/sources/Reserve.move:28]: Explain +1 (truncating arithmetic?).

File: language/documentation/examples/experimental/rounding-error/sources/Reserve.move [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+        ensures global<Coin1Info>(ADMIN).total_value == old(global<Coin1Info>(ADMIN).total_value) + amount_to_mint;
+        ensures global<Coin1Info>(ADMIN).reserve_coin2.backing_value == old(global<Coin1Info>(ADMIN).reserve_coin2.backing_value) + backing_coin2;
+        ensures backing_coin2 == coin2_amount_to_reserve;
+    }
+
+    public fun mint_coin1_incorrect(amount_to_mint: u64, backing_coin2: u64): u64 // returns the minted Coin1.
+    acquires Coin1Info {
+        assert!(amount_to_mint > 0, 1);
+        let coin1info = borrow_global_mut<Coin1Info>(ADMIN);
```
Comment (DavidLDill) [inline language/documentation/examples/experimental/rounding-error/sources/Reserve.move:43]: Explain error -- truncating arithmetic?

File: language/documentation/examples/experimental/rounding-error/sources/Uniswap.move [hunk 4]
```diff
@@ -0,0 +1,130 @@
+module NamedAddr::Uniswap {
```
Comment (DavidLDill) [inline language/documentation/examples/experimental/rounding-error/sources/Uniswap.move:1]: Explain example and where it comes from (error in Uniswap 1 -- URL to explanation?)

**Review by DavidLDill — APPROVED** [review]

**Comment by junkil-park** [comment]
@DavidLDill, thank you for your comments. I will address your comments in the next PR since I am going to add more code comments and document to the modules.

**Pull request merged**